pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -205,24 +224,59 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterf
 		// discovery call if the resources appeared in-between the calls. In that
 		// case, the restMapper will fail to map some of newResources until the next
 		// sync period.
-		if err := gc.resyncMonitors(newResources); err != nil {
-			utilruntime.HandleError(fmt.Errorf("failed to sync resource monitors: %v", err))
-			return
+		if err := gc.resyncMonitors(currentResources); err != nil {
+			return map[schema.GroupVersionResource]struct{}{}, fmt.Errorf("failed to sync resource monitors: %v", err)
 		}
-		// TODO: WaitForCacheSync can block forever during normal operation. Could
```

Comment:
Removing this TODO because this PR will stop WaitForCacheSync from blocking forever.